### PR TITLE
Find all test on a directory

### PR DESCRIPTION
**Pull request by Chemaclass**
## 📚 Description

Issue: https://github.com/TypedDevs/bashunit/issues/139

## 🔖 Changes

- Allow running `./bashunit` over a full directory
- Update docs

### CODE REVIEW

File: src/helpers.sh [hunk 1]
```diff
@@ -100,3 +100,13 @@ function helper::unsetIfExists() {
...
+
+helpers::find_files_recursive() {
+    local path="$1"
+
+    if [[ -d "$path" ]]; then
+        find "$path" -type f -name '*test.sh' | sort | uniq
+    else
+        echo "$path"
+    fi
+}
```
Comment (Tito-Kati) [inline src/helpers.sh:112]:
```suggestion
helpers::find_files_recursive() {
  local path="$1"

  if [[ -d "$path" ]]; then
    find "$path" -type f -name '*test.sh' | sort | uniq
  else
    echo "$path"
  fi
}
```

File: src/helpers.sh [hunk 2]
```diff
@@ -100,3 +100,13 @@ function helper::unsetIfExists() {
     unset "$function_name"
   fi
 }
+
+helpers::find_files_recursive() {
+    local path="$1"
+
+    if [[ -d "$path" ]]; then
+        find "$path" -type f -name '*test.sh' | sort | uniq
```
Comment (Tito-Kati) [inline src/helpers.sh:108]:
```suggestion
        find "$path" -type f -name '**/*test.sh' | sort | uniq
```

File: docs/getting-started.md [hunk 3]
```diff
@@ -100,7 +100,11 @@ Once **bashunit** is installed, you're ready to get started.
 
     You can copy and execute the following command from the root of your project if you installed **bashunit** as a Git submodule:
     ```bash
-    bashunit/bashunit tests/example_test.sh
+    # all tests inside the tests directory
+    ./bashunit tests
```
Comment (Tito-Kati) [inline docs/getting-started.md:104]: I think the Test file names section should be updated too.

**Review by Tito-Kati — CHANGES_REQUESTED** [review]

File: src/helpers.sh (hunk 2 above)
Comment (Chemaclass) [reply src/helpers.sh]: Not needed. The `find ... -name '*test.sh' ` is already looking at all files that has a name ending with `test.sh`, so we don't need to use the `**/*test.sh`. Or did you experience any issue testing it on Windows/Linux, @Tito-Kati ?

Comment (Tito-Kati) [reply src/helpers.sh]: Only the ones directly under that directory, do we want recursive search?

Comment (Chemaclass) [reply src/helpers.sh]: Isn’t `find` already searching recursively?
Screenshot 2023-10-03 at 15 18 27

**Review by JesusValera — APPROVED** [review]

File: src/helpers.sh (hunk 2 above)
Comment (Tito-Kati) [reply src/helpers.sh]: You are right xD

**Pull request merged**